tracee/tracee.go

```diff
@@ -812,6 +812,41 @@ func (t *Tracee) shouldProcessEvent(e RawEvent) bool {
 
 func (t *Tracee) processEvent(ctx *context, args map[argTag]interface{}) error {
 	switch ctx.EventID {
+
+	//capture written files
+	case VfsWriteEventID, VfsWritevEventID:
+		if t.config.CaptureWrite {
+			filePath, ok := args[t.EncParamName[ctx.EventID%2]["pathname"]].(string)
+			if !ok {
+				return fmt.Errorf("error parsing security_bprm_check args")
+			}
+			// path should be absolute, except for e.g memfd_create files
+			if filePath == "" || filePath[0] != '/' {
+				return nil
+			}
+
+			destinationDirPath := filepath.Join(t.config.OutputPath, strconv.Itoa(int(ctx.MntID)))
+			if err := os.MkdirAll(destinationDirPath, 0755); err != nil {
+				return err
+			}
+			sourceFileStat, err := os.Stat(filePath)
+			if err != nil {
+				return err
+			}
+
+			sourceFileInode := sourceFileStat.Sys().(*syscall.Stat_t).Ino
+			sourceFileDev := sourceFileStat.Sys().(*syscall.Stat_t).Dev
```

Comment:
The file's inode number ('inode' argument name) and device id ('dev' argument name) are given as part of the vfs_write(v) event itself, no need to stat for that.

Reply:
Yes!  This is what I was struggling with, thank you for the clarification.  I saw with capturing executed files, the `security_bprm_check` events had a `pathname` argument name.  Where would I find the fields for each event type?

e.g. `filePath, ok := args[t.EncParamName[ctx.EventID%2]["pathname"]].(string)`

Reply:
Have a look in consts.go.
The table named EventIdsToParams will tell you exactly that